File: php/features.php
<?php
declare(strict_types=1);

// CustomsWindow SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CustomsWindowFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CustomsWindowBaseFeature();
            case "test":
                return new CustomsWindowTestFeature();
            default:
                return new CustomsWindowBaseFeature();
        }
    }
}

File: py/features.py
# CustomsWindow SDK feature factory

from feature.base_feature import CustomsWindowBaseFeature
from feature.test_feature import CustomsWindowTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CustomsWindowBaseFeature(),
        "test": lambda: CustomsWindowTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
